extensions/shared/blocks/cover/utils.js

```diff
@@ -1,19 +1,59 @@
 
-export const videoFileExtensions = [
-	'ogv',
-	'mp4',
-	'm4v',
-	'mov',
-	'qt',
-	'wmv',
-	'avi',
-	'mpeg',
-	'mpg',
-	'mpe',
-	'3gp',
-	'3gpp',
-	'3g2',
-	'3gp2',
-	'3gp',
-	'3g2',
-];
+/**
+ * External dependencies
+ */
+import { pickBy, flatten, map, keys, values } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { isSimpleSite } from "../../site-type-utils";
+import getJetpackExtensionAvailability from "../../get-jetpack-extension-availability";
+
+/**
+ * Check if the given file is a video.
+ *
+ * @param   {string|object} file - file to check.
+ * @returns {boolean}       True if it's a video file. Otherwise, False.
+ */
+export function isVideoFile( file ) {
+	// Pick up allowed mime types from the window object.
+	const allowedMimeTypes = window?.Jetpack_Editor_Initial_State?.allowedMimeTypes;
+	if ( ! allowedMimeTypes ) {
+		return false;
+	}
+
+	if ( ! file ) {
+		return false;
+	}
+
+	let allowedVideoMimeTypes = pickBy( allowedMimeTypes, ( type ) => /^video\//.test( type ) );
+	const allowedVideoFileExtensions = flatten( map( keys( allowedVideoMimeTypes ), ext => ext.split( '|' ) ) );
+
+	if ( typeof file === 'string' ) {
+		const fileExtension = ( file.split( '.' ) )?.[ 1 ];
+		return fileExtension && allowedVideoFileExtensions.includes( fileExtension );
+	}
+
+	allowedVideoMimeTypes = values( allowedVideoMimeTypes );
```

Comment:
This appears to be divorced from the declaration of `let allowedVideoMimeTypes`. Any reason we don't just do this as a single operation and use `const`? Also why are they no collocated in the code?

Reply:
yes, the reason is that when I create `allowedVideoFileExtensions` constant I need the data collection. Once I get it, then and cover `allowedVideoMimeTypes` to an Array of string items.